app/src/main/java/com/poupa/vinylmusicplayer/loader/SongLoader.java

```diff
@@ -92,14 +115,18 @@ private static Song getSongFromCursorImpl(@NonNull Cursor cursor) {
         final long dateModified = cursor.getLong(7);
 
         // search in the discog cache first
-        Song song = Discography.getInstance().getSong(id);
-        if (song != null) {
+        Discography discog = Discography.getInstance();
```

Comment:
Please move the Discography instance to a class attribute so you're not calling getInstance again and again

Reply:
will do